lib/util/mailables.js

```diff
@@ -0,0 +1,154 @@
+  {fieldName: 'zip', placeholder: 'Zip'}
+]
+
+export function createLetter (formData, mailablesConfig) {
+  // This is a very goofy approach to convert an image URL to its image data for
+  // writing to the PDF, but it seems to be a solid approach.
+  const img = new Image()
+  img.crossOrigin = 'anonymous'
+  img.onload = () => {
+    var canvas = document.createElement('canvas')
+    canvas.width = img.width
+    canvas.height = img.height
+
+    const ctx = canvas.getContext('2d')
+    ctx.drawImage(img, 0, 0)
+
+    const data = canvas.toDataURL('image/png')
+    const imageData = {data, height: img.height, width: img.width}
+    writePDF(formData, imageData, mailablesConfig)
+  }
+  img.src = mailablesConfig.header_graphic
+}
+
+function writePDF (formData, imageData, otpConfig) {
+  const {
+    address1 = '',
+    address2,
+    city = '',
+    firstname = '',
+    lastname = '',
+    mailables,
+    state = '',
+    zip = ''
+  } = formData
+  const {
+    horizontal_margin: horizontalMargin,
+    vertical_margin: verticalMargin
+  } = otpConfig
+  const margins = {
+    top: verticalMargin,
+    bottom: verticalMargin,
+    left: horizontalMargin,
+    right: horizontalMargin
+  }
+  console.log(margins)
+  const doc = new PDFDocument({margins})
+  const stream = doc.pipe(blobStream())
+  preparePage(doc, imageData, otpConfig)
+  doc.on('pageAdded', () => preparePage(doc, otpConfig))
+
+  // current date
+  doc.text(moment().format('MMMM D, YYYY'))
+
+  // recipient address
+  doc.moveDown()
+    .moveDown()
+    .moveDown()
+    .text(firstname.toUpperCase() + ' ' + lastname.toUpperCase())
+    .text(address1.toUpperCase())
+  if (address2 && address2.length > 0) {
+    doc.text(address2.toUpperCase())
+  }
+  doc.text(city.toUpperCase() + ', ' + state.toUpperCase() + ' ' + zip)
+
+  // introduction block
+  doc.moveDown()
+    .moveDown()
+    .text(otpConfig.introduction)
+
+  // table header
+  doc.font('Helvetica-Bold')
+    .moveDown()
+    .moveDown()
+    .text('SUMMARY BY ITEM', {align: 'center'})
+    .moveDown()
+    .moveDown()
+
+  const tableData = {
+    headers: ['Item', 'Quantity'],
+    rows: mailables.map(mailable => {
+      let {name, largeFormat, largePrint, quantity} = mailable
+      if (largePrint && largeFormat) {
+        name += ' (LARGE PRINT)'
+      }
+      return [name, quantity]
+    })
+  }
+
+  doc.table(tableData, {
+    prepareHeader: () => doc.font('Helvetica-Bold'),
+    prepareRow: (row, i) => doc.font('Helvetica').fontSize(12)
+  })
+
+  // conclusion block
+  doc.moveDown()
+    .moveDown()
+    .font('Helvetica')
+    .text(otpConfig.conclusion)
+
+  doc.end()
+  stream.on('finish', () => {
+    const url = stream.toBlobURL('application/pdf')
+    window.open(url)
+  })
+}
+
+function preparePage (doc, imageData, otpConfig) {
+  const {
+    footer,
+    header_graphic: headerGraphic,
+    header_graphic_height: headerGraphicHeight,
+    header_graphic_width: headerGraphicWidth
```

Comment:
Is it possible to change these config values to be camelCase instead of snake_case?